### ISSUE TITLE
Elegit hangs when remote repo is unavailable

### ISSUE THREAD

**Issue by dmusican** [dmusican/Elegit#490]
I'm sitting in a place with public free wifi that is blocking some ports. Git push on https fails due to the port being blocked. Elegit eventually times out, but it took a very long time (a full minute, perhaps?). Git push at the command line hangs also for a long time, but you can always ctrl-c out of it. Elegit needs some kind of cancel if push is hung.

**Comment by erictwalker18** [comment]
Having trouble recreating this timing out. However, I did find that it might be caused by having multiple remotes, as the popup window to show the remotes is a threading mess and difficult to bring to the foreground.
On a repo that has one remote, does it time out slowly when all internet is blocked (I found this to time out in ~1 second) or just when some ports are blocked? Either way, we should have a ctrl-c option.

**Comment by dmusican** [comment]
Agreed. Getting this right is undoubtedly tied into the threading overhaul I'm working on.

**Comment by erictwalker18** [comment]
I'll wait to pull request until after the threading overhaul. I fixed the issue with multiple remotes in hang-fix, but haven't found a proper way to stop a push/kill the working thread mid execution unless it is looping through something.
